src/Moryx.Products.Management/Implementation/Storage/ProductStorage.cs

```diff
@@ -353,6 +353,28 @@ public IReadOnlyList<IProductType> LoadTypes(ProductQuery query)
             else if (query.RecipeFilter == RecipeFilter.WithoutRecipes)
                 productsQuery = productsQuery.Where(p => p.Recipes.Count == 0);
 
+            // Exclude parts
+            if (query.NoPartLinks)
+            {
```

Comment:
What's the point of this? The ProductsQuery never includes PartLinks in the response. It only returns the head type for the query arguments and then you need to call LoadType on each.

Copying all properties onto a new object within a LinQ query seems like bad taste. And it would break the selectors 5 lines below.

If there is a runtime issue with proxies or lazy loading, check the UnitOfWork settings at the top or the conversion at the bottom.